app/code/Magento/ImportExport/Test/Unit/Model/Import/Source/FileFactoryTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\ImportExport\Test\Unit\Model\Import\Source;
+
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Filesystem;
+use Magento\ImportExport\Model\Import\Source\FileFactory;
+use Magento\ImportExport\Model\Import\Source\FileParser\CsvParserFactory;
+use Magento\ImportExport\Model\Import\Source\FileParser\UnsupportedPathException;
+use Magento\ImportExport\Test\Unit\Model\Import\Source\FileParser\FakeObjectManager;
+use Magento\ImportExport\Test\Unit\Model\Import\Source\FileParser\FakeParser;
+use Magento\ImportExport\Test\Unit\Model\Import\Source\FileParser\FakeParserFactory;
+
+class FileFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGivenParserFactoryEmptyWhenSourceIsCreatedByPathThenNoSourceIsCreated()
+    {
+        $this->setExpectedException(UnsupportedPathException::class, 'Path "test.csv" is not supported');
+
+        $fileFactory = new FileFactory(
+            new FakeParserFactory(),
+            new FakeObjectManager()
+        );
+
+        $fileFactory->createFromFilePath('test.csv');
+    }
+
+    public function testGivenParserFactoryConfiguredWhenSourceIsCreatedByPathThenRightInstanceIsCreated()
```

Comment:
Method name looks like this because "_" are prohibited by Magento code style guideline. Previous versions had "_" as space.